modules/component/internals/linkedList.ts

```diff
@@ -1,20 +1,35 @@
+/**
+ * Immutable linked list data structure implementation
+ */
 export abstract class LinkedList<A> {
   abstract readonly head: A
   abstract readonly tail: LinkedList<A>
 
   abstract isEmpty: boolean
+  /**
+   * Prepends node to the linked list
+   * @param element Node to be added
+   */
   prepend(element: A): LinkedList<A> {
     return new Cons(element, this)
   }
 
   static get empty(): LinkedList<never> {
     return new Empty()
   }
-
+  /**
+   * Lifts an element as node in a linked list
+   * @param element Element to be lifted as node in linked list
+   */
   static of<T>(element: T): LinkedList<T> {
     return new Cons<T>(element, LinkedList.empty)
   }
 
+  /**
+   * Execute a fn on each element of the list, resulting in a single output value.
+   * @param seed A value to use as the second argument to the first call of the fn.
```

Comment:
should be called `accumulator` imo, to keep inline with people's existing intuition about `reduce`